lib/pbench/server/api/resources/datasets_compare.py

```diff
@@ -96,6 +96,7 @@ def _get(self, params: ApiParams, req: Request, context: ApiContext) -> Response
 
         datasets = params.query.get("datasets")
         benchmark_choice = None
+        benchmark = None
         for dataset in datasets:
```

Comment:
While initializing `benchmark` here certainly doesn't hurt anything, I don't see how it helps anything, either:  it is written unconditionally at line 113 before it is tested at line 116, so the initial value is always overwritten (and, `benchmark` is never read if we don't enter the loop).

Reply:
`vscode` was unreasonably paranoid about the reference outside the loop, and I pre-declared the thing just to shut it up.

Reply:
Curious:  I didn't see the reference outside the loop (otherwise, I probably would have ratified the paranoia).  Still, I don't object to shutting up the linter (I think that's the better alternative to trying to ignore it).

Reply:
Oh, [here](https://github.com/dbutenhof/pbench/blob/66101b19994cdfb34764f7661400f23168f6c63e/lib/pbench/server/api/resources/datasets_compare.py#L155) it is.  Yeah, that reference probably should have been changed to `benchmark_choice`, rather than expanding the scope of `benchmark`....